From 546cda6519ccfb4025dca90af63c0eb1ec6f9d45 Mon Sep 17 00:00:00 2001
From: Shadi Ammouri <shadi@marvell.com>
Date: Thu, 5 May 2016 09:03:43 +0300
Subject: [PATCH 140/239] llc: Remove LLC status check before llc invalidate

This is incorrect, as we want to invalidate the LLC before we enable it.

Change-Id: I39ec850d576e8d4e3fd948717292b66479ca0a7d
Signed-off-by: Shadi Ammouri <shadi@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29510
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
Reviewed-by: Yoav Gvili <ygvili@marvell.com>
Tested-by: Yoav Gvili <ygvili@marvell.com>
---
 drivers/marvell/cache_llc.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/marvell/cache_llc.c b/drivers/marvell/cache_llc.c
index 2919524..05df902 100644
--- a/drivers/marvell/cache_llc.c
+++ b/drivers/marvell/cache_llc.c
@@ -50,8 +50,6 @@ void llc_clean_all(void)
 
 void llc_inv_all(void)
 {
-	/* Invalidating when L2 is enabled is a nono */
-	assert(mmio_read_32(MVEBU_LLC_BASE + LLC_CTRL) & LLC_CTRL_EN);
 	mmio_write_32(MVEBU_LLC_BASE + L2X0_INV_WAY, LLC_WAY_MASK);
 	llc_cache_sync();
 }
-- 
1.9.1

